### ISSUE TITLE
Integrate with react-hook-form `control`

### ISSUE THREAD

**Issue by hanza93** [eduardoborges/use-mask-input#59]
We are using the controlled version for react-hook-form by using `useControl` hook. it would be awesome if we can use `useHookFormMask` with `control` too

**Comment by alysonvilela** [comment]
You can use it. Simply re-subscribe ref value using withMask function

```js
<Controller
  name="test"
  render={({ field }) => {
    return <input {...field} ref={withMask('9999-9999')}/>; // ✅
  }}
/>
```

**Comment by eduardoborges** [comment]
Hey @hanza93 im working on it, but for now you can use `withMask` like @alysonvilela demostrate above. <3

**Comment by adrianwilk89** [comment]
@alysonvilela is problem with this solution because if you pass withMask as ref then primary functionality that scroll and focus element with errors stop working i mean if you set `shouldFocusError` to `true`

**Comment by wendellpbarreto** [comment]
True, @adrianwilk89. I'm facing this problem right now, my inputs are not being focused anymore.

**Comment by wendellpbarreto** [comment]
I'm currently using the solution below, it's fully working for now, I don't know yet if it has any side effects.

```js
<Controller
  name="test"
  render={({ field }) => {
    return <input {...field} ref={ref => {
              withMask('9999-9999')(ref)
              field.ref(ref)
            }}  /> 
  }}
/>
```
